### PR TITLE
Lightning mode refactor and disable/enable qp mode toggle 

### PR DESCRIPTION
**Pull request by minhtuev**
## What changes are proposed in this pull request?

Refactored lightning mode to rename as query performance mode and allowed enabling/disabling qp mode toggle through deployment vars 

## How is this patch tested? If it is not, please explain why.

* Manual test (✅ )
* Loom video: https://www.loom.com/share/9b5acd7984e74b2aa7a68a194b8fb37d?sid=d0fdcfe6-2426-4a0c-9439-6a4488af906f

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Introduced new selectors for query performance configuration.

- **Improvements**
	- Renamed and restructured components related to performance modes for clarity and consistency.
	- Updated tooltip texts and links to reflect the new query performance mode.
	- Enhanced rendering logic to ensure correct display of performance-related information.

- **Documentation**
	- Updated component exports and type definitions for better clarity.

### CODE REVIEW

**Comment by sashankaryal** [comment]
Lightning mode e2e tests are failing. It might be an actual regression or the specs might have to be updated.

File: app/packages/core/src/components/Actions/Options.tsx [hunk 1]
```diff
@@ -216,19 +216,20 @@ const DynamicGroupsViewMode = ({ modal }: { modal: boolean }) => {
...
 };
 
-const Lightning = () => {
+const QueryPerformance = () => {
   const [threshold, setThreshold] = useRecoilState(fos.lightningThreshold);
   const config = useRecoilValue(fos.lightningThresholdConfig);
   const reset = useResetRecoilState(fos.lightningThreshold);
   const count = useRecoilValue(fos.datasetSampleCount);
   const theme = useTheme();
+  const enableQpMode = useRecoilValue(fos.enableQueryPerformanceConfig) && useRecoilValue(fos.defaultQueryPerformanceConfig);
```
Comment (benjaminpkane) [inline app/packages/core/src/components/Actions/Options.tsx:225]: In React, hooks [cannot be run conditionally](https://dev.to/lexlohr/conditional-hooks-1fpj). This should be changed to two separate lines

Comment (minhtuev) [reply app/packages/core/src/components/Actions/Options.tsx]: Done ✅

**Review by benjaminpkane — APPROVED** [review]
LGTM 🚀

**Pull request merged**